Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -194,6 +219,56 @@ class BrowserViewController: UIViewController {
         didInit()
         
         rewards.delegate = self
+        
+        if rewards.ledger.isEnabled && !rewards.ledger.isWalletCreated {
+            rewards.createWalletIfNeeded()
+        }
+    }
+    
+    static func legacyWallet(for config: BraveRewardsConfiguration) -> BraveLedger? {
+        let fm = FileManager.default
+        let stateStorage = URL(fileURLWithPath: config.stateStoragePath)
+        let legacyLedger = stateStorage.appendingPathComponent("legacy_ledger")
+
+        // Check if we've already migrated the users wallet to the `legacy_rewards` folder
+        if fm.fileExists(atPath: legacyLedger.path) {
+            BraveLedger.environment = config.environment
+            return BraveLedger(stateStoragePath: legacyLedger.path)
+        }
+        
+        // We've already performed an attempt at migration, if there wasn't a legacy folder, then
+        // we have no legacy wallet.
+        if Preferences.Rewards.migratedLegacyWallet.value {
+            return nil
+        }
+        
+        // Ledger exists in the state storage under `ledger` folder, if that folder doesn't exist
+        // then the user hasn't actually launched the app before and doesn't need to migrate
+        let ledgerFolder = stateStorage.appendingPathComponent("ledger")
+        if !fm.fileExists(atPath: ledgerFolder.path) {
+            // No wallet, therefore no legacy folder needed
+            Preferences.Rewards.migratedLegacyWallet.value = true
+            return nil
+        }
+        
+        do {
+            // Copy the current `ledger` directory into the new legacy state storage path
+            try fm.copyItem(at: ledgerFolder, to: legacyLedger)
+            // Remove the old Rewards DB so that it starts fresh
+            try fm.removeItem(atPath: ledgerFolder.appendingPathComponent("Rewards.db").path)
+            // And remove the sqlite journal file if it exists
+            let journalPath = ledgerFolder.appendingPathComponent("Rewards.db-journal").path
```

Comment:
We duplicate the users wallet then reset activity info + BAT in current wallet so that AC doesn't use the users legacy BAT. If we deleted entire folder the user would have to create a new wallet, enable rewards & ads again, etc.